counter/src/main/java/org/infinispan/counter/impl/weak/WeakCounterImpl.java

```diff
@@ -62,9 +61,10 @@
  * @author Pedro Ruivo
  * @since 9.0
  */
-public class WeakCounterImpl implements WeakCounter, CounterEventGenerator, TopologyChangeListener {
+public class WeakCounterImpl implements WeakCounter, CounterEventGenerator, TopologyChangeListener, InternalCounterAdmin {
+
+   private static final AtomicReferenceFieldUpdater<Entry, Long> SNAPSHOT_UPDATER = AtomicReferenceFieldUpdater.newUpdater(Entry.class, Long.class, "snapshot");
```

Comment:
Cause I assume `init` is always invoked before using the `Entry` ?

Reply:
unfortunately no :( 
The listener is registered first so you can have an event set the value before `init` is invoked. That's why I have a `Long` object field. If it is not null, `init` does nothing. 

If I register the lister after `init`, it could miss some updates from other nodes and the local value would be out of sync.

Reply:
Alright, thanks for explanation.